sklearn/inspection/__init__.py

```diff
@@ -1,8 +1,21 @@
 """The :mod:`sklearn.inspection` module includes tools for model inspection."""
-from ._partial_dependence import partial_dependence
-from ._partial_dependence import plot_partial_dependence
-from ._partial_dependence import PartialDependenceDisplay
-from ._permutation_importance import permutation_importance
+
+# TODO: remove me in 0.24 (as well as the noqa markers)
+# pre-cache the import of the deprecated module so that import
+# sklearn.inspection.partial_dependence returns the function as in
+# 0.21, instead of the module
+# https://github.com/scikit-learn/scikit-learn/issues/15842
+import warnings
+with warnings.catch_warnings():
+    warnings.simplefilter("ignore", category=FutureWarning)
+    from . import partial_dependence as _ignored
+    del _ignored
+
+from ._partial_dependence import partial_dependence  # noqa
```

Comment:
For simplicity we can:

```py
with warnings.catch_warnings():
    warnings.simplefilter("ignore", category=FutureWarning)
    from .partial_dependence import partial_dependence
```

which will emulate the behavior in 0.21.

Reply:
Alright.